eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/steps/VariantsAnnotGenerateInputBatch.java

```diff
@@ -0,0 +1,190 @@
+        reader.setFields("{ chr : 1, start : 1, end : 1, ref : 1, alt : 1, type : 1}");
+        reader.setTargetType(DBObject.class);
+        reader.setTemplate(template);
+
+        Map<String, Sort.Direction> coordinatesSort = new HashMap<>();
+        coordinatesSort.put("chr", Sort.Direction.ASC);
+        coordinatesSort.put("start", Sort.Direction.ASC);
+        reader.setSort(coordinatesSort);
+        return reader;
+    }
+
+    @Bean
+    public ItemProcessor<DBObject, VariantWrapper> vepInputLineProcessor() {
+        return new VariantAnnotationItemProcessor();
+    }
+
+    /**
+     * @return must return a {@link FlatFileItemWriter} and not a {@link org.springframework.batch.item.ItemWriter}
+     * {@see https://jira.spring.io/browse/BATCH-2097
+     *
+     * TODO: The variant list should be compressed
+     */
+    @Bean
+    public FlatFileItemWriter<VariantWrapper> vepInputWriter() throws Exception {
+        return initWriter(new FileSystemResource(pipelineOptions.getString("vepInput")));
+    }
+
+    public static FlatFileItemWriter<VariantWrapper> initWriter(Resource resource){
+        BeanWrapperFieldExtractor<VariantWrapper> fieldExtractor = new BeanWrapperFieldExtractor<>();
+        fieldExtractor.setNames(new String[] {"chr", "start", "end", "refAlt", "strand"});
+
+        DelimitedLineAggregator<VariantWrapper> delLineAgg = new DelimitedLineAggregator<>();
+        delLineAgg.setDelimiter("\t");
+        delLineAgg.setFieldExtractor(fieldExtractor);
+
+        FlatFileItemWriter<VariantWrapper> writer = new FlatFileItemWriter<>();
+
+        writer.setResource(resource);
+        writer.setAppendAllowed(false);
+        writer.setShouldDeleteIfExists(true);
+        writer.setLineAggregator(delLineAgg);
+        return writer;
+    }
+
+    @Bean
+    public MongoOperations mongoOperations() {
+        MongoTemplate mongoTemplate;
+        try {
+            mongoTemplate = getMongoTemplate();
+        } catch (UnknownHostException e) {
+            throw new RuntimeException("Unable to initialize MongoDB", e);
+        }
+        return mongoTemplate;
+    }
+
+    private MongoTemplate getMongoTemplate() throws UnknownHostException {
+        MongoTemplate mongoTemplate;
+        if(pipelineOptions.getString("dbAuthenticationDb").isEmpty()){
+            mongoTemplate = ConnectionHelper.getMongoTemplate(
+                    pipelineOptions.getString(VariantStorageManager.DB_NAME)
+            );
+        }else {
+            mongoTemplate = ConnectionHelper.getMongoTemplate(
+                    pipelineOptions.getString("dbHosts"),
+                    pipelineOptions.getString("dbAuthenticationDb"),
+                    pipelineOptions.getString(VariantStorageManager.DB_NAME),
+                    pipelineOptions.getString("dbUser"),
+                    pipelineOptions.getString("dbPassword").toCharArray()
+            );
+        }
+        return mongoTemplate;
+    }
```

Comment:
`mongoOperations()` and `getMongoTemplate()` methods present here as well as in `ConnectionHelper`. Should they be extracted (and the class renamed to `MongoConnectionHelper`)?